.github/workflows/deploy.yml

```diff
@@ -0,0 +1,57 @@
+name: Deploy
+
+on:
+  pull_request_target:
+    branches: 
+      - main
+    types: [ closed ]
+
+jobs:
+  validate:
+    uses: "./.github/workflows/validate.yml"
+
+  # GitHub Actions does not have a halt job option to stop from deploying if no functional changes were found.
+  # We build a separate job to substitute the halt option.
+  # The `deploy` job is dependent on the output of the `check-for-functional-changes` job.
+  check-for-functional-changes:
+    runs-on: ubuntu-22.04
+    if: github.event.pull_request.merged == true
+    outputs:
+      status: ${{ steps.stop-early.outputs.status }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Fetch all the tags
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9.12
+      - id: stop-early
+        run: if "${GITHUB_WORKSPACE}/.github/has-functional-changes.sh" ; then echo "::set-output name=status::success" ; fi
+
+  deploy:
+    runs-on: ubuntu-22.04
+    needs: [ validate, check-for-functional-changes ]
+    if: github.event.pull_request.merged == true && needs.check-for-functional-changes.outputs.status == 'success'
+    env:
+      PYPI_TOKEN_OPENFISCA_BOT: ${{ secrets.PYPI_TOKEN_OPENFISCA_BOT }}
```

Comment:
Do we really need to set this variable in `env`, can't we directly interpolate `${{ secrets.PYPI_TOKEN_OPENFISCA_BOT }}` in the upload step? 🙂